.github/workflows/windows.yml

```diff
@@ -55,45 +53,44 @@ jobs:
-      - name: export custom recipes
+          conan profile update settings.compiler.runtime=MT${{ matrix.configuration == 'Debug' && 'd' || '' }} default
+          # Do not quote the URL. An empty string will be accepted (with
+          # a non-fatal warning), but a missing argument will not.
+          conan remote add ripple ${{ env.CONAN_URL }} --insert 0
+      - name: try to authenticate to ripple Conan remote
+        shell: bash
+        id: remote
         run: |
-          conan export external/snappy snappy/1.1.9@
-          conan export external/soci soci/4.0.3@
-      - name: install dependencies
+          echo outcome=$(conan user --remote ripple ${{ secrets.CONAN_USERNAME }} --password ${{ secrets.CONAN_TOKEN }} && echo success || echo failure) | tee ${GITHUB_OUTPUT}
+      - name: list missing binaries
+        id: binaries
+        shell: bash
+        # Print the list of dependencies that would need to be built locally.
+        # A non-empty list means we have "failed" to cache binaries remotely.
         run: |
-          mkdir $env:build_dir
-          cd $env:build_dir
-          conan install .. --build missing --settings build_type=${{ matrix.configuration }}
-      - name: configure
-        run: |
-          $env:build_dir
-          cd $env:build_dir
-          pwd
-          ls
-          cmake `
-            -G "${{ matrix.generator }}" `
-            -DCMAKE_TOOLCHAIN_FILE:FILEPATH=build/generators/conan_toolchain.cmake `
-            -Dassert=ON `
-            -Dreporting=OFF `
-            -Dunity=OFF `
-            ..
+          echo missing=$(conan info . --build missing --settings build_type=${{ matrix.configuration }} --json 2>/dev/null  | grep '^\[') | tee ${GITHUB_OUTPUT}
+      - name: build dependencies
+        if: (steps.binaries.outputs.missing != '[]')
+        uses: ./.github/actions/dependencies
+        with:
+          configuration: ${{ matrix.configuration }}
+      - name: upload dependencies to remote
+        if: (steps.binaries.outputs.missing != '[]') && (steps.remote.outputs.outcome == 'success')
+        run: conan upload --remote ripple '*' --all --parallel --confirm
       - name: build
-        run: |
-          cmake --build $env:build_dir --target rippled --config ${{ matrix.configuration }} --parallel $env:NUMBER_OF_PROCESSORS
+        uses: ./.github/actions/build
+        with:
+          generator: '${{ matrix.generator }}'
+          configuration: ${{ matrix.configuration }}
+          # Hard code for now. Move to the matrix if varied options are needed
+          cmake-args: '-Dassert=ON -Dreporting=OFF -Dunity=OFF'
       - name: test
+        shell: bash
         run: |
-          & "$env:build_dir\${{ matrix.configuration }}\rippled.exe" --unittest
+          # ${build_dir}/${{ matrix.configuration }}/rippled --unittest --unittest-jobs $(nproc)
```

Comment:
Is `nproc` available on Windows? @ximinez please confirm, but I think parallel testing works on Windows. I believe we just have to use `$NUMBER_OF_PROCESSORS`. If neither works, though, we should leave a comment here.

Reply:
`nproc` is available. I commented out this line to see if the tests would pass on a single threaded run on the Github runners. [They didn't.](https://github.com/ximinez/rippled/actions/runs/6340590156/job/17222329753#step:14:1372)

Instead what I'm going to do is put the parallel run back, but mark the test step as allowed to fail (`continue-on-error`). At this point, it's more important to me that the build succeeds than that the tests succeed.
1. We've got plenty of test coverage on the other jobs.
2. Test failures are much rarer than build failures because of cross-platform issues.
3. Having a test failure locally doesn't interrupt *my* workflow nearly as much as a build failure.